src/test/java/com/ihren/task2/exercise1/ExceptionHandlerTest.java

```diff
@@ -0,0 +1,159 @@
+package com.ihren.task2.exercise1;
+
+import com.ihren.task2.exercise1.exception.ApplicationException;
+import com.ihren.task2.exercise1.exception.MappingException;
+import com.ihren.task2.exercise1.handler.DLQHandler;
+import com.ihren.task2.exercise1.model.Metrics;
+import io.vavr.control.Try;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.BDDMockito.willDoNothing;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(MockitoExtension.class)
+class ExceptionHandlerTest {
+
+    @InjectMocks
+    private ExceptionHandler exceptionHandler;
+
+    @Mock
+    private Metrics metrics;
+
+    @Mock
+    private DLQHandler dlqHandler;
+
+    private static Stream<Arguments> dataProvider() {
+        return Stream.of(
+                Arguments.of("test", 4),
+                Arguments.of(4, "test"),
+                Arguments.of(List.of(0, 0), 2),
+                Arguments.of(2, List.of(0, 0)),
+                Arguments.of(null, null)
+        );
+    };
+
+    @ParameterizedTest
+    @MethodSource("dataProvider")
+    <I, E> void should_ReturnOK_when_InputIsValid( I input, E expected) {
```

Comment:
These parameters are configured without sense because you always mock them. You should have just one healthy test for this cse